finagle-core/src/test/scala/com/twitter/finagle/filter/RequestMeterFilterTest.scala

```diff
@@ -0,0 +1,44 @@
+package com.twitter.finagle.filter
+
+import com.twitter.concurrent.AsyncMeter
+import com.twitter.conversions.time._
+import com.twitter.finagle.util.DefaultTimer
+import com.twitter.finagle.{Failure, Service}
+import com.twitter.util.{Await, Future}
+import org.junit.runner.RunWith
+import org.scalatest.FunSuite
+import org.scalatest.junit.JUnitRunner
+
+@RunWith(classOf[JUnitRunner])
+class RequestMeterFilterTest extends FunSuite {
+
+  val timer = DefaultTimer.twitter
+  val meter = AsyncMeter.newMeter(1, 1 second, 1)(timer)
+
+  test("return service execution after getting a permit") {
+    val echoSvc = new Service[Int, Int] {
+      def apply(req: Int) = Future(req)
+    }
+    val svc = new RequestMeterFilter(meter) andThen echoSvc
+    assert(Await.result(svc(2)) == 2)
+  }
+
+  test("mark dropped requests as rejected") {
+    val neverSvc = new Service[Int, Int] {
+      def apply(req: Int) = Future.never
+    }
+    val svc = new RequestMeterFilter(meter, 2) andThen neverSvc
+    val f = intercept[Failure] { Await.result(svc(1)) }
+    assert(f.isFlagged(Failure.Restartable))
+  }
+
+  test("service failures are not wrapped as rejected") {
+    val exc = new Exception("app exc")
```

Comment:
Not sure I got the idea of using `MockTimer`. It proceeds with `svc(1)` succesfully but when it calls `svc(2)` it gets stuck on the `meter.await(1)` part. It hangs forever...

```scala
  test("mark dropped requests as rejected") {
    val timer = new MockTimer
    val neverSvc = new Service[Int, Int] {
      def apply(req: Int) = Future.never
    }
    Time.withCurrentTimeFrozen { ctl =>
      val meter = AsyncMeter.newMeter(1, 1 second, 1)(timer)
      val svc = new RequestMeterFilter(meter) andThen neverSvc
      svc(1)

      val f = intercept[Failure] { Await.result(svc(2)) }
      assert(f.isFlagged(Failure.Restartable))

      ctl.advance(1 second)
      timer.tick()
    }
  }
```